public/static/js/admin.js

```diff
@@ -87,6 +88,28 @@ mciModule.controller('AdminSettingsController', ['$scope','$window', 'mciAdminRe
       return;
     }
 
+    try {
+      $scope.Settings.container_pools.pools = jsyaml.safeLoad($scope.tempContainerPools);
+    } catch(e) {
+      alert("Error parsing container pools yaml: " + e);
+      return;
+    }
+
+    if ($scope.tempContainerPools === null || $scope.tempContainerPools === undefined || $scope.tempContainerPools == "") {
+      $scope.Settings.container_pools.pools = [];
+    }
+
+    // do not save settings if duplicate container pool IDs found
+    var uniqueIds = new Set()
+    for (var i = 0; i < $scope.Settings.container_pools.pools.length; i++) {
```

Comment:
I would do this validation before setting $scope.Settings.container_pools.pools above. Also you can check for unique IDs by constructing an object, and setting its keys as you encounter them